File: services/ai-service/src/routers/sentence.py
"""AI 造句评估 Router"""
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from services.sentence_agent import evaluate_sentence
from utils.logger import logger

router = APIRouter(prefix="/sentence", tags=["sentence"])


class SentenceEvalRequest(BaseModel):
    word: str
    meaning_cn: str
    sentence: str


class SentenceEvalResponse(BaseModel):
    grammar_score: int
    usage_score: int
    natural_score: int
    overall_score: int
    feedback: str
    improved_sentence: str


@router.post("/evaluate", response_model=SentenceEvalResponse)
async def evaluate(request: SentenceEvalRequest):
    """评估用户造句"""
    try:
        result = await evaluate_sentence(
            word=request.word,
            meaning_cn=request.meaning_cn,
            user_sentence=request.sentence,
        )
        return result
    except Exception as e:
        logger.error(f"造句评估失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))

File: apps/mobile/src/api/modules/sentence.ts
/**
 * AI 造句挑战 API
 */
import { Client } from '../client';
import { ENDPOINTS } from '../endpoints';

export interface SentenceEvalRequest {
  word: string;
  wordId: string;
  meaningCn: string;
  sentence: string;
}

export interface SentenceEvalResponse {
  grammarScore: number;
  usageScore: number;
  naturalScore: number;
  overallScore: number;
  feedback: string;
  improvedSentence: string;
}

declare module '../client' {
  interface Client {
    evaluateSentence(data: SentenceEvalRequest): Promise<SentenceEvalResponse>;
  }
}

/**
 * 提交造句获取 AI 评估
 */
Client.prototype.evaluateSentence = async function(data: SentenceEvalRequest): Promise<SentenceEvalResponse> {
  return this.api.post(ENDPOINTS.WORDS_SENTENCE_CHALLENGE, data);
};
